src/projection_client.rs

```diff
@@ -0,0 +1,629 @@
+    settings: ClientSettings,
+}
+
+impl ProjectionClient {
+    pub async fn create(settings: ClientSettings) -> Result<Self, Box<dyn std::error::Error>> {
+        let client = GrpcClient::create(settings.clone()).await?;
+
+        Ok(ProjectionClient { client, settings })
+    }
+
+    pub async fn create_one_time_projection(
+        &self,
+        query: String,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()> {
+        self.create_projection(
+            credentials,
+            projections::create_req::Options {
+                query,
+                mode: Some(projections::create_req::options::Mode::OneTime(Empty {})),
+            },
+        )
+        .await
+    }
+
+    pub async fn create_transient_projection<Name>(
+        &self,
+        name: Name,
+        query: String,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        self.create_projection(
+            credentials,
+            projections::create_req::Options {
+                query,
+                mode: Some(projections::create_req::options::Mode::Transient(
+                    projections::create_req::options::Transient {
+                        name: name.as_ref().to_string(),
+                    },
+                )),
+            },
+        )
+        .await
+    }
+
+    pub async fn create_continuous_projection<Name>(
+        &self,
+        name: Name,
+        query: String,
+        track_emitted_streams: bool,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        self.create_projection(
+            credentials,
+            projections::create_req::Options {
+                query,
+                mode: Some(projections::create_req::options::Mode::Continuous(
+                    projections::create_req::options::Continuous {
+                        name: name.as_ref().to_string(),
+                        track_emitted_streams,
+                    },
+                )),
+            },
+        )
+        .await
+    }
+
+    async fn create_projection(
+        &self,
+        credentials: Option<&Credentials>,
+        options: projections::create_req::Options,
+    ) -> crate::Result<()> {
+        let req = projections::CreateReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel);
+                let _ = client.create(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn update_projection<Name>(
+        &self,
+        name: Name,
+        query: String,
+        emit_option: Option<bool>,
```

Comment:
**CONCERN:** Shouldn't it be called `track_emitted_streams` as in the `Create` method?

Reply:
I believe is not the same option. In this case, it's about enabling or disabling emit of events from the projection, not tracking. I'll double-check with the server.